src/main/java/com/atlassian/db/replica/internal/ConnectionParameters.java

```diff
@@ -36,15 +38,23 @@ public void initialize(Connection connection) throws SQLException {
         }
     }
 
-    public void setTransactionIsolation(Integer transactionIsolation) {
+    public void setTransactionIsolation(
+        Supplier<Optional<Connection>> currentConnection,
```

Comment:
In the current form, `ConnectionParameters` could contain the `state` and set the transaction if any raw connection initialized. I could refactor it now, but `state` needs to be refactored anyway. I don't want to enter the rabbit-hole of the refactoring 😨 .

Reply:
Fair call